netlink-packet-route/src/rtnl/tc/message.rs

```diff
@@ -1,31 +1,69 @@
 // SPDX-License-Identifier: MIT
+use std::convert::From;
 
 use anyhow::Context;
 
 use crate::{
-    nlas::tc::Nla,
-    traits::{Emitable, Parseable},
-    DecodeError,
-    TcMessageBuffer,
-    TC_HEADER_LEN,
+    constants::*,
+    nlas::{
+        self,
+        tc::{Nla, Stats, Stats2, StatsBuffer},
+        DefaultNla, NlaBuffer, NlasIterator,
+    },
+    parsers::{parse_string, parse_u8},
+    traits::{Emitable, Parseable, ParseableParametrized},
+    DecodeError, TcMessageBuffer, TC_HEADER_LEN,
 };
 
-#[derive(Debug, PartialEq, Eq, Clone, Default)]
-pub struct TcMessage {
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct TcMessage<A> {
     pub header: TcHeader,
-    pub nlas: Vec<Nla>,
+    pub nlas: Vec<Nla<A>>,
 }
 
-impl TcMessage {
-    pub fn into_parts(self) -> (TcHeader, Vec<Nla>) {
+impl<A> TcMessage<A> {
+    pub fn into_parts(self) -> (TcHeader, Vec<Nla<A>>) {
         (self.header, self.nlas)
     }
 
-    pub fn from_parts(header: TcHeader, nlas: Vec<Nla>) -> Self {
+    pub fn from_parts(header: TcHeader, nlas: Vec<Nla<A>>) -> Self {
         TcMessage { header, nlas }
     }
 }
 
+impl<A> Default for TcMessage<A> {
+    fn default() -> Self {
+        Self {
+            nlas: Vec::new(),
+            ..Default::default()
+        }
+    }
+}
+
+impl<A> From<i32> for TcMessage<A> {
```

Comment:
I feel like this `From` impl is not needed. We can simply do:
```rust
let mut tc_msg = TcMessage<Class>::default();
tc_msg.header.index = 1;
```
Also it's not obvious what the relation between an i32 and a TcMessage is. One has to look at the code to realize it corresponds to the index.